### DISCUSSION TITLE
REST API PUT on /process-instance/{process-instance-id}/variables returns with 413 when payload exceeds 100KiB

### DISCUSSION THREAD

**Discussion by PretoPlasma** [camunda-community-hub/community#107] [Q&A] [answered]
Hello community,
I work with camunda engine since 2021. In a POC i tried to figure out the limits of the Camunda REST API.
I created a process with exact one user task (so the instance will never end unless the user task is completed) to use it as a storage for key/value-pairs.
I defined a variable "data" keeping those pairs as a JSON.
When putting JSON into this variable via camunda cockpit manually it seems that I could put a very huge JSON into it.
When putting JSON into this variable using the REST API PUT on /process-instance/{process-instance-id}/variables it returns with 413 when the payload exceeds 100KiB.
I am aware of the recommendations by camunda not to store too much data into process instances (https://docs.camunda.io/docs/components/concepts/variables/#variable-size-limitation) however I was wondering where this limit comes from as I did not find any restrictions in the deployment / configuration of the spring boot based camunda run I use in a docker environment.
Does anyone can explain me where this 100KiB limit comes?
Looking forward to any reply.
Kind regards,
P.

**Comment by xomiamoore** [comment, marked as answer]
Hi @PretoPlasma! This GitHub Discussions space is designed for questions around the Camunda Community Hub and the various repos located here.

Please post your question in the [Camunda forum instead.](https://forum.camunda.io/c/camunda-platform-8-topics/40) Thank you!
